Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -235,13 +237,13 @@ void DiscusMultipleScatteringCorrection::exec() {
   }
   const MatrixWorkspace &instrumentWS = useSparseInstrument ? *sparseWS : *inputWS;
 
-  auto instrument = inputWS->getInstrument();
+  m_instrument = inputWS->getInstrument();
   const auto nhists = useSparseInstrument ? static_cast<int64_t>(sparseWS->getNumberHistograms())
                                           : static_cast<int64_t>(inputWS->getNumberHistograms());
 
-  const auto &sample = inputWS->sample();
+  m_sampleShape = std::shared_ptr<const Geometry::IObject>(inputWS->sample().getShape().clone());
```

Comment:
Is there a way to simply get the shared_ptr without cloning? That way we hold the lifetime too but don't have to do a potentially expensive copy

Reply:
Equally, if we hold the sptr for inputWs will that indirectly keep the lifetime for the underlying shape?
We might be able to hold onto a ref and rely on the fact we've pinned the `inputWs` to manage lifetime :thinking:

Reply:
There's another nasty looking clone() in the MonteCarloAbsorption algorithm (in MCInteractionVolume.cpp) which is probably someone else trying to work around the same problem. I think I'll just add an extra getShapePtr method onto the Sample class so I can (separately) fix that up as well